staging_vespalib/src/vespa/vespalib/util/singleexecutor.cpp

```diff
@@ -98,8 +124,7 @@ SingleExecutor::run_tasks_till(uint64_t available) {
         task->run();
         _rp.store(++consumed, std::memory_order_release);
         if (wakeupLimit == consumed) {
-            MonitorGuard guard(_producerMonitor);
-            guard.broadcast();
+            wakeupProducer();
```

Comment:
note that this signal may be lost since no state is being passed with the signal (the other thread might call wait after the signal has been discarded since no-one was waiting).

Reply:
Yes, correct. The purpose is to wake up one that is sleeping waiting for low watermark to be reached.

Reply:
The purpose is irrelevant if the signal is lost. The thread you are trying to wake up will keep sleeping. This might not happen too often, but it could be a bit surprising when it does.

Reply:
Now it will be attempted woken up again after every 100us. I think that should be good enough.